### PR TITLE
Expanding rings guides on automation through spk and adding clarifica…

### LINKED ISSUE #1112: Walkthrough for rings management via spk
**As a:**
spk user
**I want:**
A walkthrough guide that shows how and when a user would call spk ring create <ring-name>,spk ring set-default <ring-name>, and finally spk ring delete <ring-name>
**So that:**
I can understand why, when, and how to use rings via spk, and what the downstream artifacts in my HLD and final manifests would be.

**Acceptance Criteria:**
- [x] Walkthrough on rings through spk
  - [x] How to add a ring
  - [x] How to remove a ring
  - [x] Setting the default ring and what does it do
  - [x] Pipeline clarification: There only ever needs to be 1 lifecycle per application repo. It will point to the default branch defined in the rings to look for `bedrock.yaml` as the source of truth.
    - related question to on which branch the lifecycle pipeline should trigger off of: https://github.com/microsoft/bedrock/issues/1143
  - [x] Pipeline clarification: There only needs to be 1 build pipeline for a service. Adding a ring will add the pipeline branch trigger to the branch pipeline.

**Additional context:**
manual guide to rings: https://github.com/CatalystCode/spk/blob/master/guides/manual-guide-to-rings.md
rings 101: https://github.com/CatalystCode/spk/blob/master/guides/rings-101.md

This walkthrough should also drive #974 smoke tests for rings.

**Does this require updates to documentation?:**
Yes

### PR DESCRIPTION
**Pull request by mtarng**
…tion on pipelines

Closes: https://github.com/microsoft/bedrock/issues/1112

Also opened up this question about our lifecycle pipeline: https://github.com/microsoft/bedrock/issues/1143

### CODE REVIEW

**Comment by mtarng** [comment]
@edaena please give me feedback for the walkthrough portions via the rings101 guide, because you're looking at the smoke tests for rings. If you think there needs to be an explicit guide with full step by step and results let me know.

**Comment by mtarng** [comment]
@pjlittle @kjeffc this is the PR for the updated rings guides.

**Review by edaena — APPROVED** [review]

**Review by yradsmikham — APPROVED** [review]

File: guides/manual-guide-to-rings.md [hunk 1]
```diff
@@ -276,20 +276,25 @@ variables:
 4. Observe that committing to the `test-new-feature` should no longer trigger
    builds.
 
-_Do note that deleting a `ring` presently does not remove the service and `ring`
-from a cluster. This work is being
-[tracked here](https://github.com/microsoft/bedrock/issues/858) but the
-following instructions will detail how this can be done manually_
+**Note:** Deleting a `ring` presently does not remove the service and `ring`
+from a cluster as the project lifecycle pipeline does not yet remove rings or
+services from the HLD repository.\_ The work to support the automated removal of
```
Comment (evanlouie) [inline guides/manual-guide-to-rings.md]: should the `\_` be there?

File: guides/rings-101.md [hunk 2]
```diff
@@ -89,6 +108,41 @@ This command removes the ring from your `bedrock.yaml` file and walks through
 all the services in your project and removing the ring branch from the service
 pipeline YAML.
 
+**Note:** The "default" ring cannot be deleted. If you wish to remove the ring
+defined under `bedrock.yaml` with `isDefault: true`, you must first set another
+ring to be the default ring via `spk ring set-default <new-default-ring-name>`.
+
+**Note:** Deleting a `ring` presently does not remove the service and `ring`
+from a cluster as the project lifecycle pipeline does not yet remove rings or
+services from the HLD repository.\_ The work to support the automated removal of
```
Comment (evanlouie) [inline guides/rings-101.md]: should the `\_` be there?

**Review by evanlouie — APPROVED** [review]

**Pull request merged**